### ISSUE TITLE
Notifications: how to disable native and use "chrome"-style instead?

### ISSUE THREAD

**Issue by vadimyer** [electron/electron#5299]
* Electron version: 0.37.7
* Operating system: Windows 10

So that's pretty much the whole question in topic.

Currently I got the notification system built in my app that's being opened as an external URL, but this way I don't know how to handle on click events (so when the window is hidden it's shown up when clicked on) - since the code is in the app, not main.js.

Also, is it possible to disable sounds for native notifications and use HTML5 audio api instead, custom sounds?

Sorry, I'm new to Electron.

**Comment by MarshallOfSound** [comment]
> Currently I got the notification system built in my app that's being opened as an external URL, but this way I don't know how to handle on click events (so when the window is hidden it's shown up when clicked on) - since the code is in the app, not main.js.

This doesn't make much sense to me, can you rephrase?

> Also, is it possible to disable sounds for native notifications and use HTML5 audio api instead, custom sounds?

Silent Notification --> https://developer.mozilla.org/en-US/docs/Web/API/Notification/silent
Custom Sounds --> https://developer.mozilla.org/en-US/docs/Web/API/notification/sound

**Comment by vadimyer** [comment]
As for the first question, I've already managed to handle onclick event through IPC messages. I needed to call show() event on browser window if it's minimized, by clicking on a notification (the code from renderer can't call the code from main so...). Didn't find a better solution.

As for sounds, great! I'll test it out.

**Comment by MarshallOfSound** [comment]
IPC messages would probably be the best way to go.  There are however methods such as 

```js
require('remote').getCurrentWindow().show();
```

Which will work if the window you want to show is the window that launched the notification.

Or if the window you want to show is stored as a global variable in the main process you can call

```js
require('remote').getGlobal('variableName').show();
```

**Comment by vadimyer** [comment]
Thanks.

Seems like custom sounds aren't supported yet in Windows? Cause again it's an HTML5 API that's noted in docs only is supported from renderer.

**Comment by MarshallOfSound** [comment]
Check if it is supported in the latest Chrome

**Comment by vadimyer** [comment]
Ok I'm blind, "sound" property doesn't work in Chrome as stated on that page, but silent - works. So I've disabled the system's notification sound and just went with audio.play(). Flawless!

**Comment by zcbenz** [comment]
On the original question, we only implemented native notifications, and we don't have plan to port the chrome-style one.
